src/charm.py

```diff
@@ -182,6 +189,45 @@ def _on_config_changed(self, _: ops.HookEvent, charm_state: CharmState) -> None:
         self.change_config(charm_state)
         self._set_workload_version()
 
+    @inject_charm_state
+    def _on_relation_joined(self, _: ops.HookEvent, charm_state: CharmState) -> None:
+        """Handle Synapse peer relation joined event.
+
+        Args:
+            charm_state: The charm state.
+        """
+        if charm_state.redis_config is None:
+            logger.debug("Scaling is not allowed without Redis integration.")
+            self.model.unit.status = ops.BlockedStatus(
+                "Redis integration not found. Please, verify it."
+            )
+            return
+        self.model.unit.status = ops.ActiveStatus()
+
+    @inject_charm_state
+    def _on_relation_departed(self, event: RelationDepartedEvent, charm_state: CharmState) -> None:
+        """Handle Synapse peer relation departed event.
+
+        Args:
+            event: relation departed event.
+            charm_state: The charm state.
+        """
+        if not self.unit.is_leader() or event.departing_unit == self.unit:
```

Comment:
Not sure I get the `event.departing_unit == self.unit` part

Reply:
This check is because all units including the one departing receive this event. Thats why the charm will ignore it if not the leader or if is the unit removed/departing.

Reference:
https://juju.is/docs/sdk/relation-name-relation-departed-event#heading--observing-this-event-in-ops

Reply:
Shouldn't it be the other way around? `if not self.unit.is_leader() or event.departing_unit != self.unit`

Reply:
Oh, I see. Makes sense to me